### PR TITLE
backport: Merge bitcoin#24237, 23631, 24609

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -3651,7 +3651,11 @@ void PeerManagerImpl::ProcessMessage(
 
         int64_t nTimeOffset = nTime - GetTime();
         pfrom.nTimeOffset = nTimeOffset;
-        AddTimeData(pfrom.addr, nTimeOffset);
+        if (!pfrom.IsInboundConn()) {
```
Comment (knst) [inline src/net_processing.cpp:3654]: pick also bitcoin/bitcoin#24609 which update documentation also; this 2 backports should be merged together or one-after-one

Comment (vijaydasmp) [reply src/net_processing.cpp:3654]: added bitcoin#24609

**Review by knst — DISMISSED** [review]
utACK 7fb9ac2d8544ba5462d9887f15fd6bb300e6f88d

File: src/init.cpp [hunk 2]
```diff
@@ -570,7 +570,7 @@ void SetupServerArgs(ArgsManager& argsman)
...
     argsman.AddArg("-maxreceivebuffer=<n>", strprintf("Maximum per-connection receive buffer, <n>*1000 bytes (default: %u)", DEFAULT_MAXRECEIVEBUFFER), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
     argsman.AddArg("-maxsendbuffer=<n>", strprintf("Maximum per-connection memory usage for the send buffer, <n>*1000 bytes (default: %u)", DEFAULT_MAXSENDBUFFER), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
-    argsman.AddArg("-maxtimeadjustment", strprintf("Maximum allowed median peer time offset adjustment. Local perspective of time may be influenced by peers forward or backward by this amount. (default: %u seconds)", DEFAULT_MAX_TIME_ADJUSTMENT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
+    argsman.AddArg("-maxtimeadjustment", strprintf("Maximum allowed median peer time offset adjustment. Local perspective of time may be influenced by outbound peers forward or backward by this amount. (default: %u seconds)", DEFAULT_MAX_TIME_ADJUSTMENT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```
Comment (UdjinM6) [inline src/init.cpp:573]: 24609:
```suggestion
    argsman.AddArg("-maxtimeadjustment", strprintf("Maximum allowed median peer time offset adjustment. Local perspective of time may be influenced by outbound peers forward or backward by this amount (default: %u seconds).", DEFAULT_MAX_TIME_ADJUSTMENT), ArgsManager::ALLOW_ANY, OptionsCategory::CONNECTION);
```

**Review by PastaPastaPasta — COMMENTED** [review]
generally LGTM 7fb9ac2d8544ba5462d9887f15fd6bb300e6f88d

**Review by UdjinM6 — APPROVED** [review]
utACK e237301b76b923b2c44c81ae40b7e31f5a1db429

**Review by knst — APPROVED** [review]
LGTM e237301b76b923b2c44c81ae40b7e31f5a1db429

**Pull request merged**